internal/controllers/httpsedge_controller.go

```diff
@@ -643,6 +645,71 @@ func (u *edgeRouteModuleUpdater) setEdgeRouteResponseHeaders(ctx context.Context
 	return err
 }
 
+func (u *edgeRouteModuleUpdater) setEdgeRouteOAuth(ctx context.Context, route *ngrok.HTTPSEdgeRoute, routeSpec *ingressv1alpha1.HTTPSEdgeRouteSpec) error {
+	oauth := routeSpec.OAuth
+	oauthClient := u.clientset.OAuth()
+
+	if oauth == nil {
+		if route.OAuth == nil {
+			u.log.Info("OAuth matches desired state, skipping update")
+			return nil
+		}
+
+		return oauthClient.Delete(ctx, u.edgeRouteItem(route))
+	}
+
+	var module *ngrok.EndpointOAuth
+	var err error
+
+	providers := []OAuthProvider{
+		oauth.Google,
+		oauth.Github,
+		oauth.Gitlab,
+		oauth.Amazon,
+		oauth.Facebook,
+		oauth.Microsoft,
+		oauth.Twitch,
+		oauth.Linkedin,
+	}
+
+	for _, p := range providers {
+		if p == nil {
+			continue
+		}
+
+		var secret *string
+		secretKeyRef := p.ClientSecretKeyRef()
+
+		// Look up the client secret key if its specified,
+		// otherwise default to nil
+		if secretKeyRef != nil {
+			secret, err = u.getSecret(ctx, *secretKeyRef)
+			if client.IgnoreNotFound(err) != nil {
```

Comment:
do you we need to handle the not found case here as well?

Reply:
You are right. I missed this in the refactor. We should probably get rid of the `client.IgnoreNotFound` around the `err`. If you specify a secret and its not found, we should return it rather than trying to keep going

Reply:
Fixed in 9ebd898